Added additional sample projects.

MoFtZ opened the pull request:
Added fixed size buffer example for #597.

QUESTION: Can you think of any other sample projects we should add?

m4rs-mt commented:
@MoFtZ Maybe we can add an additional sample demonstrating the use of our `Random` functionality:

- Using the high-level `Random` API to fill a buffer with uniformly distributed floats (can also include the `CuRand` features)
- Using the kernel `Random` API while accessing a data view inside a GPU kernel to generate random numbers on the fly

MoFtZ commented:
@m4rs-mt I'm trying to use the `RNGView` within a kernel, but it raises an exception on the CPU accelerator inside `GetRandomProvider`. Did you have any suggestions for how to fix the issue?

MoFtZ commented:
Fixed issue with `RNGView` on the CPU accelerator. On the CPU accelerator, the group size was not always the same as the warp size. And even on Cuda accelerators, my group size was 30, compared to a warp size of 32, so it also had potential issues.

Also updated the `SimpleViews` sample to show offset copying to fix https://github.com/m4rs-mt/ILGPU/issues/604.

m4rs-mt commented on Src/ILGPU/Runtime/ArrayViewExtensions.cs:
```diff
@@ -1209,6 +1209,40 @@ public static void MemSetToZero<TView>(this TView view)
+        /// <param name="source">The source view instance.</param>
+        /// <param name="span">The CPU data target.</param>
+        /// <remarks>This method is not supported on accelerators.</remarks>
+        [NotInsideKernel]
```
nice extension methods 👍
